compiler/src/dotty/tools/dotc/transform/TupleRewrites.scala

```diff
@@ -0,0 +1,215 @@
+package dotty.tools.dotc
+package transform
+
+import ast.Trees._
+import core.Constants.Constant
+import core.Contexts.Context
+import core.Definitions.MaxImplementedTupleArity
+import core.StdNames._
+import core.Symbols._
+import core.Types._
+import transform.TreeTransforms.{MiniPhaseTransform, TransformerInfo}
+
+/** Local rewrites for tuple expressions. Nested apply and unapply trees coming
+ *  from desugaring into single apply/unapply nodes on DottyTupleN/LargeTuple.
+ */
+class TupleRewrites extends MiniPhaseTransform {
```

Comment:
I did not understand why we need DottyTuple in addition to scala.TupleN classes. It would be good to document this.